apollo-api/src/main/java/com/spotify/apollo/RequestValue.java

```diff
@@ -88,7 +88,9 @@ public Request withHeader(String name, String value) {
   @Override
   public Request withHeaders(Map<String, String> additionalHeaders) {
     Map<String, String> headers = new LinkedHashMap<>(headers());
-    headers.putAll(additionalHeaders);
+    for (Map.Entry<String, String> header : additionalHeaders.entrySet()) {
+      headers.put(header.getKey().toLowerCase(), header.getValue());
+    }
```

Comment:
I don't know the relevance. In #71 I suggest that service implementations provide their own implementations of `Request` to deal with this.